### PR TITLE
[#1067] Disable logging for production

### LINKED ISSUE #1067: Disable logging for production
We log a lot of information and that is potentially dangerous. We need to redact many struct, like #1056 or #1063, etc.

In the end we decided to log as less information as possible. 

### The goal
Disable logging whenever possible but for production only. This will very likely influence also the UI, with no logs, we can and should hide button `export logs only` in Export private data screen.

### PR DESCRIPTION
**Pull request by LukasKorba**
Closes #1067 

- There are 3 different and independent loggers that have been disabled for production (release configuration builds)
- The TCA logger doesn't call os_log()
- The SDK is instantiated with .noLogging policy
- The Wallet logger is never set so it's nil
- With no logs the export logs button in Export Private Data screen no longer makes sense so I removed it from the UI

This code review checklist is intended to serve as a starting point for the author and reviewer, although it may not be appropriate for all types of changes (e.g. fixing a spelling typo in documentation).  For more in-depth discussion of how we think about code review, please see [Code Review Guidelines](../blob/main/CODE_REVIEW_GUIDELINES.md).

# Author

- [x] Self-review: Did you review your own code in GitHub's web interface? Code often looks different when reviewing the diff in a browser, making it easier to spot potential bugs.
- [ ] Does the code abide by the [Coding Guidelines](../blob/main/docs/CODING_GUIDELINES.md)?
- [ ] Automated tests: Did you add appropriate automated tests for any code changes?
- [x] Code coverage: Did you check the code coverage report for the automated tests?  While we are not looking for perfect coverage, the tool can point out potential cases that have been missed.
- [ ] Documentation: Did you update Docs as appropiate? (E.g [README.md](../blob/main/README.md), etc.)
- [x] Run the app: Did you run the app and try the changes? 
- [ ] Did you provide Screenshots of what the App looks like before and after your changes as part of the description of this PR? (only applicable to UI Changes)
- [x] Rebase and squash: Did you pull in the latest changes from the main branch and squash your commits before assigning a reviewer? Having your code up to date and squashed will make it easier for others to review. Use best judgement when squashing commits, as some changes (such as refactoring) might be easier to review as a separate commit.

# Reviewer

- [ ] Checklist review: Did you go through the code with the [Code Review Guidelines](../blob/main/CODE_REVIEW_GUIDELINES.md) checklist?
- [ ] Ad hoc review: Did you perform an ad hoc review?  _In addition to a first pass using the code review guidelines, do a second pass using your best judgement and experience which may identify additional questions or comments. Research shows that code review is most effective when done in multiple passes, where reviewers look for different things through each pass._
- [ ] Automated tests: Did you review the automated tests?
- [ ] Manual tests: Did you review the manual tests?_You will find manual testing guidelines under our [manual testing section](../blob/main/docs/testing/manual_testing)_
- [ ] How is Code Coverage affected by this PR? _We encourage you to compare coverage before and after your changes and when possible, leave it in a better place. [Learn More...](../blob/main/docs/testing/local_coverage.md)_
- [ ] Documentation: Did you review Docs, [README.md](../blob/main/README.md), [LICENSE.md](../blob/main/LICENSE.md), and [Architecture.md](../blob/main/docs/Architecture.md) as appropriate?
- [ ] Run the app: Did you run the app and try the changes? While the CI server runs the app to look for build failures or crashes, humans running the app are more likely to notice unexpected log messages, UI inconsistencies, or bad output data.

### CODE REVIEW

**Review by nuttycom — APPROVED** [review]
utACK

File: secant.xcodeproj/xcshareddata/xcschemes/secant-mainnet.xcscheme [hunk 1]
```diff
@@ -31,7 +31,7 @@
       </Testables>
    </TestAction>
    <LaunchAction
-      buildConfiguration = "Debug"
+      buildConfiguration = "Release"
```
Comment (str4d) [inline secant.xcodeproj/xcshareddata/xcschemes/secant-mainnet.xcscheme:34]: I don't know if this is an intentional change, or a side-effect of reconfiguring something in XCode for testing this PR.

File: secant.xcodeproj/xcshareddata/xcschemes/secant-mainnet.xcscheme [hunk 2]
```diff
@@ -50,6 +50,12 @@
             ReferencedContainer = "container:secant.xcodeproj">
          </BuildableReference>
       </BuildableProductRunnable>
+      <CommandLineArguments>
+         <CommandLineArgument
+            argument = "-FIRDebugEnabled"
+            isEnabled = "YES">
```
Comment (str4d) [inline secant.xcodeproj/xcshareddata/xcschemes/secant-mainnet.xcscheme:56]: Likewise, I don't know if this is an intentional change.

**Review by str4d — APPROVED** [review]
utACK f7862a3082724db5235f79c11d4c2e189a208d57

File: secant.xcodeproj/xcshareddata/xcschemes/secant-mainnet.xcscheme (hunk 1 above)
Comment (LukasKorba) [reply secant.xcodeproj/xcshareddata/xcschemes/secant-mainnet.xcscheme]: Good catch, not ideal to leave release. Reverted.

**Pull request merged**